scipy/signal/filter_design.py

```diff
@@ -3855,16 +3856,30 @@ def iirpeak(w0, Q):
     # Compute -3dB atenuation
     gb = 1/np.sqrt(2)
 
-    # Compute beta: formula 11.3.19 (p.579) from reference [1]
-    beta = (gb/np.sqrt(1.0-gb**2.0))*np.tan(bw/2.0)
+    if ftype == "notch":
+        # Compute beta: formula 11.3.4 (p.575) from reference [1]
+        beta = (np.sqrt(1.0-gb**2.0)/gb)*np.tan(bw/2.0)
 
-    # Compute gain: formula 11.3.20 (p.579) from reference [1]
-    gain = 1.0/(1.0+beta)
+        # Compute gain: formula 11.3.6 (p.575) from reference [1]
+        gain = 1.0/(1.0+beta)
```

Comment:
To make it clearer which things are different, why not have a second conditional like:
```
if ftype == 'notch':
    beta = ...
elif ...
else ...
gain = 1.0/(1.0 + beta)
a = np.array([1, -2*gain*np.cos(w0), 2*gain - 1
if ftype == 'notch':
    b = ...
else:
    b = ...
return b, a
...
```
WDYT?